libs/sdk-ui-geo/src/GeoPushpinChart.tsx

```diff
@@ -106,14 +127,41 @@ function GeoPushpinChartInner(props: IGeoPushpinChartProps): JSX.Element {
 
 const WrappedGeoPushpinChart = withTheme(withContexts(GeoPushpinChartInner));
 
-/**
- * @public
- */
-export const GeoPushpinChart = (props: IGeoPushpinChartProps) => {
+const LocationGeoPushpinChart = (props: ILocationGeoPushpinChartProps) => {
     const [location, size, color, segmentBy, filters, sortBy] = useResolveValuesWithPlaceholders(
         [props.location, props.size, props.color, props.segmentBy, props.filters, props.sortBy],
         props.placeholdersResolutionContext,
     );
-
     return <WrappedGeoPushpinChart {...props} {...{ location, size, color, segmentBy, filters, sortBy }} />;
 };
+
+const LatitudeLongitudeGeoPushpinChart = (props: ILongitudeLatitudeGeoPushpinChartProps) => {
+    const [longitude, latitude, size, color, segmentBy, filters, sortBy] = useResolveValuesWithPlaceholders(
+        [
+            props.longitude,
+            props.latitude,
+            props.size,
+            props.color,
+            props.segmentBy,
+            props.filters,
+            props.sortBy,
+        ],
+        props.placeholdersResolutionContext,
+    );
+    return (
+        <WrappedGeoPushpinChart
+            {...props}
+            {...{ longitude, latitude, size, color, segmentBy, filters, sortBy }}
+        />
+    );
+};
+
+/**
+ * @public
+ */
+export const GeoPushpinChart = (props: IGeoPushpinChartProps) => {
```

Comment:
Split into two smaller components so `useResolveValuesWithPlaceholders` hook can be called conditionally based on prop type.